src/mpid/ch4/shm/ipc/gpu/gpu_init.c

```diff
@@ -75,9 +75,7 @@ int MPIDI_GPU_mpi_init_hook(int rank, int size, int *tag_bits)
     MPIR_CHKPMEM_DECL(1);
 
     MPIDI_GPUI_global.initialized = 0;
-    mpl_err = MPL_gpu_init(&device_count, &my_max_dev_id);
-    MPIR_ERR_CHKANDJUMP(mpl_err != MPL_SUCCESS, mpi_errno, MPI_ERR_OTHER, "**gpu_init");
-
+    MPL_gpu_get_dev_count(&device_count, &my_max_dev_id);
```

Comment:
I don't think there's reason to remove the error checking here. Can you add it back?